spyder/plugins/editor/widgets/tests/test_introspection.py

```diff
@@ -764,6 +767,51 @@ def test_fallback_completions(fallback_codeeditor, qtbot):
     code_editor.toggle_code_snippets(True)
 
 
+@pytest.mark.slow
+@pytest.mark.first
+@flaky(max_runs=5)
+def test_kite_textEdit_completions(mock_codeeditor, qtbot):
+    """Test on-the-fly completions."""
+    code_editor, mock = mock_codeeditor
+    completion = code_editor.completion_widget
+
+    code_editor.toggle_automatic_completions(False)
+    code_editor.toggle_code_snippets(False)
+
+    # Set cursor to start
+    code_editor.go_to_line(1)
+
+    qtbot.keyClicks(code_editor, 'my_dict.')
```

Comment:
Why the `.` at end of `my_dict` is necessary here?

Reply:
The reason I use this test case is because this is a type of completion that Kite returns (it allows users to quickly access dict keys with a `.`), and it's a case where the completion `["dict-key"]` replacing the `.` can't be represented with just `insertText`.

Reply:
Ok, I didn't know that and it's pretty cool!